api-specifications.md

```diff
@@ -96,6 +104,161 @@ Create a course (anyone logged in). Used for outside Exchange.
 * Login required
 * Course already exists
 
+### /api/instructor: Course instructor management
+
+#### POST /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Add an instructor to the course. (instructors only)*
+
+##### Request (HTTP POST data)
+```
+first_name=/*instructor name*/&
+last_name=/*instructor last name*/&
+email=/*instructor email*/
+```
+
+##### Response
+```javascript
+{
+    "success": true
+}
+```
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+
+#### GET /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Gets information about a course instructor. (instructors+students)*
+
+##### Response
+```javascript
+{
+    "success": true,
+    "first_name": /*instructor name*/,
+    "last_name": /*instructor last name*/,
+    "email": /*instructor email*/
+}
+```
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+* Instructor not found
+
+#### DELETE /api/instructor/&lt;course_id&gt;/&lt;instructor_id&gt;
+*Remove a course instructor (instructors only)*
+
+##### Error messages
+* Login required
+* Permission denied
+* Course not found
+* Instructor not found
+
```

Comment:
Add "Removing self"

Reply:
is this not covered by permission denied?

Reply:
Though depending on implementation, Permission denied is likely to only check whether the user is a course instructor.

Reply:
What's wrong with an instructor removing him/herself?

Reply:
You may get a course not accessible by anyone.

Reply:
What about preventing the last instructor from being removed? (Alternatively, delete a course with no instructors, but I don't like this idea.)

Reply:
So currently does nbgrader support multiple instructors and/or managing instructors?
If not, I think preventing last instructor removed is a good idea (though I still think if removing self, there should be a warning in frontend)

Reply:
The error message will be "Removing last instructor from course". (I am currently implementing it)